### ISSUE TITLE
[Refactor] Create shared/core/defaults.mojo for ML hyperparameter defaults

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ml-odyssey#2454]
## Problem

Default hyperparameters are scattered across multiple modules:
- Leaky ReLU alpha: `0.01` (activation.mojo)
- ELU alpha: `1.0` (activation.mojo)
- Dropout rate: `0.5` (dropout.mojo)
- BatchNorm momentum: `0.1` (layers/batchnorm.mojo)
- Augmentation probability: `0.5` (transforms.mojo)
- Text augmentation probability: `0.1-0.15` (text_transforms.mojo)
- Uniform init bounds: `-0.1` to `0.1` (initializers.mojo)
- Random seed: `42` (random.mojo)

## Files Affected

- `shared/core/activation.mojo`
- `shared/core/dropout.mojo`
- `shared/core/layers/batchnorm.mojo`
- `shared/core/initializers.mojo`
- `shared/data/transforms.mojo`
- `shared/data/text_transforms.mojo`
- `shared/utils/random.mojo`
- New: `shared/core/defaults.mojo`

## Proposed Solution

1. Create `shared/core/defaults.mojo` with ML hyperparameter defaults:
   ```mojo
   # Activation defaults
   alias DEFAULT_LEAKY_RELU_ALPHA: Float64 = 0.01
   alias DEFAULT_ELU_ALPHA: Float64 = 1.0
   alias DEFAULT_HARD_TANH_MIN: Float64 = -1.0
   alias DEFAULT_HARD_TANH_MAX: Float64 = 1.0
   
   # Regularization defaults
   alias DEFAULT_DROPOUT_RATE: Float64 = 0.5
   alias DEFAULT_BATCHNORM_MOMENTUM: Float64 = 0.1
   
   # Initialization defaults
   alias DEFAULT_UNIFORM_LOW: Float64 = -0.1
   alias DEFAULT_UNIFORM_HIGH: Float64 = 0.1
   
   # Augmentation defaults
   alias DEFAULT_AUGMENTATION_PROB: Float64 = 0.5
   alias DEFAULT_TEXT_AUGMENTATION_PROB: Float64 = 0.1
   
   # Misc
   alias DEFAULT_RANDOM_SEED: Int = 42
   ```

2. Update modules to import defaults (optional - can keep local defaults as well)

## Expected Impact

- Easier hyperparameter discovery
- Centralized documentation of defaults
- Simpler to modify common defaults

## Priority

Low (Phase 4)
